src/thor/route_action.cc

```diff
@@ -17,16 +17,114 @@ using namespace valhalla::baldr;
 using namespace valhalla::sif;
 using namespace valhalla::thor;
 
-namespace valhalla {
-namespace thor {
-
+namespace {
 // Threshold for running a second pass pedestrian route with adjusted A*. The first
 // pass for pedestrian routes is run with an aggressive A* threshold based on walking
 // speed. If ferries are included in the path the A* heuristic rules can be violated
 // which can lead to irregular paths. Running a second pass with less aggressive
 // A* can take excessive time for longer paths - so exclude them to protect the service.
 constexpr float kPedestrianMultipassThreshold = 50000.0f; // 50km
 
+/**
```

Comment:
the is the method that we call, when we want to make a uturn mid edge. it first detects that we are in a situation that allows for it, and then generates discontinuity objects (used in map matching) to pass to trip path builder, which uses those to trim the shape. there is some correlary changes on the TPB side to handle this slightly differently than the map matching request (since we dont trim in that case).